docker/linux-image.sh

```diff
@@ -171,12 +171,14 @@ main() {
         sharutils \
         gnupg
 
-    # amd64 has conflicting versions of the packages installed, so
+    # image arch has conflicting versions of the packages installed, so
     # we need to remove the system installs later. since apt relies
     # on these packages, we need to download them and reinstall
     # using dpkg later, since we cannot redownload via apt.
+    local image_arch
+    image_arch=$(dpkg --print-architecture)
```

Comment:
this is unfortunate, would be nice if it could be grabbed in "build" time for the dockerfile

Reply:
We could run a script and export it to a file during the build.

Reply:
This could also affect #952, since we also use `dpkg --print architecture` there. It would be likely a good idea then to store that information in a file if we wish to deduplicate that effort. Maybe store the information in `/opt/image-arch`?

Reply:
I think this is actually fine, since `dpkg --print-architecture` just avoids us having to use any case statements, and since this is only done once, it's quite simple and explicit. I've changed it to `dpkg_arch` as suggested below.

Reply:
> looks good, I'm just not sure I like the naming.
> maybe same-platform?

So `Dockerfile.same-platform`? Sounds good, I was just keeping it consistent with our runner naming.

> I'd also like it if with this pr that the image is made available and used when possible.

Not sure what you mean by this?

Reply:
> > I'd also like it if with this pr that the image is made available and used when possible.
> 
> Not sure what you mean by this?

This pr is only a Dockerfile, it's not built and published, and not used by cross

Reply:
It is though:


```rust

pub fn build_docker_image(
    BuildDockerImage {
        ..
    }: BuildDockerImage,
    engine: &docker::Engine,
    msg_info: &mut MessageInfo,
) -> cross::Result<()> {
    let native_dockerfile = docker_root.join("Dockerfile.native").to_utf8()?.to_string();
    let mut results = vec![];
    for (platform, (target, dockerfile)) in targets
        .iter()
        .flat_map(|t| platforms.iter().map(move |p| (p, t)))
    {
        ... 

        if target.sub.is_none() && is_native(docker_platform.as_str(), target, msg_info)? {
            dockerfile = native_dockerfile.clone();
        }

        ...
    }

    ...
}

fn is_armv6() -> bool {
    cfg!(any(
        cross_triple = "arm-unknown-linux-gnueabi",
        cross_triple = "arm-unknown-linux-musleabi"
    ))
}

fn is_armv7() -> bool {
    cfg!(any(
        cross_triple = "armv7-unknown-linux-gnueabihf",
        cross_triple = "armv7-unknown-linux-musleabihf"
    ))
}

fn is_native(
    platform: &str,
    target: &crate::ImageTarget,
    msg_info: &mut MessageInfo,
) -> cross::Result<bool> {
    Ok(match target.sub {
        Some(_) => false,
        None => match (platform, target.name.as_str()) {
            ("linux/386", "i686-unknown-linux-gnu")
            | ("linux/amd64", "x86_64-unknown-linux-gnu")
            | ("linux/arm64" | "linux/arm64/v8", "aarch64-unknown-linux-gnu")
            | ("linux/ppc64le", "powerpc64le-unknown-linux-gnu")
            | ("linux/riscv64", "riscv64gc-unknown-linux-gnu")
            | ("linux/s390x", "s390x-unknown-linux-gnu") => true,
            ("linux/arm/v6", "arm-unknown-linux-gnueabi") if is_armv6() => {
                note_host_target_detection(msg_info)?;
                true
            }
            ("linux/arm" | "linux/arm/v7", "armv7-unknown-linux-gnueabihf") if is_armv7() => {
                note_host_target_detection(msg_info)?;
                true
            }
            _ => false,
        },
    })
}
```

https://github.com/Alexhuszagh/cross/blob/3920f34c0c2b2b1a3e70ad75aeb194b4433c61a2/xtask/src/build_docker_image.rs#L191-L193

https://github.com/Alexhuszagh/cross/blob/3920f34c0c2b2b1a3e70ad75aeb194b4433c61a2/xtask/src/build_docker_image.rs#L349-L374

We detect if the image architecture is the same as the architecture the container engine is running on (IE, it is not being emulated), and use `Dockerfile.native` in that case, otherwise we use the non-native image.

Reply:
I misunderstood what this pr did :)